tools/release_builder.py

```diff
@@ -205,394 +173,283 @@ def extract_json_file(package_dict, name):
+            '\n'.join(marathon_lines).encode('utf-8')).decode()
+
+
+    def _update_resource_json(self, package_json):
+        '''Rewrites all artifact urls in pkgdir to self.release_artifact_http_dir.
+        Returns the original urls.
         '''
         # we expect the artifacts to share the same directory prefix as the stub universe file itself:
         original_artifact_prefix = '/'.join(self._stub_universe_url.split('/')[:-1])
-        logger.info('Replacing artifact prefix {} with {}'.format(
-            original_artifact_prefix, self._release_artifact_http_dir))
-        original_artifact_urls = []
-        # find all URLs, across all json files, which match the directory of the stub universe file:
-        # TODO(nickbp): once command.json is finally gone, this could just check resource.json.
-        for filename in os.listdir(pkgdir):
-            path = os.path.join(pkgdir, filename)
-            with open(path, 'r') as orig_file:
-                orig_content = orig_file.read()
-                found = re.findall('({}/.+)\"'.format(original_artifact_prefix), orig_content)
-                original_artifact_urls += found
-                new_content = orig_content.replace(original_artifact_prefix, self._release_artifact_http_dir)
-                self._update_file_content(path, orig_content, new_content)
+        log.info('Replacing artifact prefix {} with {}'.format(
+            original_artifact_prefix, self._http_directory_url))
+        # find all URLs in resource.json which match the directory of the stub universe file.
+        # update those URLs to point to the new artifact path.
+        orig_content = json.dumps(package_json['resource'], indent=2)
+        original_artifact_urls = re.findall('({}/.+)\"'.format(original_artifact_prefix), orig_content)
+        new_content = orig_content.replace(original_artifact_prefix, self._http_directory_url)
+        package_json['resource'] = json.loads(new_content, object_pairs_hook=collections.OrderedDict)
+
+        if self._release_docker_image:
+            # Find the current docker image name in resource.json, update it to the new name:
+            try:
+                docker_dict = package_json['resource']['assets']['container']['docker']
+                assert len(docker_dict) == 1
+                orig_docker_image = list(docker_dict.values())[0]
+                docker_dict[list(docker_dict.keys())[0]] = self._release_docker_image
+            except KeyError:
+                raise Exception('Release to docker specified, but no docker image found in resource.json')
+
+            # Download/reupload docker image to target name:
+            log.info('Downloading docker image {}'.format(orig_docker_image))
+            self._run_cmd('docker pull {}'.format(orig_docker_image))
+            self._run_cmd('docker tag {} {}'.format(orig_docker_image, self._release_docker_image))
+            if self._dry_run:
```

Comment:
For future, we could wrap command in a method so that we could handle the dry-runs more cleanly.